### PR TITLE
[CHORE] update dependencies, unsafe pip packages, fastapi

### PR DESCRIPTION
**Pull request by dni**
## security issues
* update fastapi / starlette because of this https://github.com/lnbits/lnbits/security/dependabot/20
* https://github.com/lnbits/lnbits/security/dependabot/18 this one is related to pyln-client depending on it. updating to latest did not help. same goes for this issue. created issue on corelightning repo, and it got into the roadmap for coming up release. https://github.com/ElementsProject/lightning/issues/6164

### parsing of url params
some of the extension are broken, see following screenshot, for latest fastapi it is not allowed anymore to Query for a url path parameter, like```("/api/v1/{name}") and myfn(name: str = Query(none))```, just use plain type.```myfn(name: str )```

screenshot-1681055593

screenshot-1681055547

- unfortunately this issue got copied across multiple extensions and will require fixing those aswell in some places.

### Request url_for
starlettes request.url_for changed the return type to `URL` instead of  `str` . you can just to `str(req.url_for())` to make the types work again.

## corelightning
update to latest pyln-client, it is also possible now to support latest corelightning node 23.02.2 because it does not break description_hash anymore

### CODE REVIEW

**Comment by dni** [comment]
trying to fix this https://github.com/lnbits/lnbits/issues/1755 here aswell

**Comment by motorina0** [comment]
> trying to fix this #1755 here aswell

- bad idea. Keep the PR small.
- are the refactorings in [lnbits/app.py](https://github.com/lnbits/lnbits/pull/1609/files#diff-e6e017cf88dd39a8ffe52eafde6f12bdc1bd8cf164a668a084dcb16a95ee925f) mandatory?
- for a PR named `update dependencies, unsafe pip packages, fastapi` I would only expect changes to:
   - `poetry.lock`, `pyproject.toml` files
   - some `imports` (maybe)

**Comment by daywalker90** [comment]
https://github.com/lnbits/lnbits/pull/1609/commits/f691d8d6fcf8d0467647743152c234fb735d5c82 works on my debian 12 btw

File: lnbits/core/views/generic.py [hunk 1]
```diff
@@ -247,11 +246,19 @@ async def wallet(
...
+    user = await get_user(usr_param)
     if not user:
         return {"status": "ERROR", "reason": "User does not exist."}
 
-    wallet = user.get_wallet(request.query_params.get("wal"))
+    wal_param = request.query_params.get("wal")
+    if not wal_param:
+        return {"status": "ERROR", "reason": "wal parameter not provided."}
+
+    wallet = user.get_wallet(wal_param)
```
Comment (jackstar12) [inline lnbits/core/views/generic.py:261]: This should be handled by a `user` and `wallet` dependency as it is used multiple times

**Comment by dni** [comment]
> > trying to fix this #1755 here aswell
> 
>     * bad idea. Keep the PR small.
> 
>     * are the refactorings in [lnbits/app.py](https://github.com/lnbits/lnbits/pull/1609/files#diff-e6e017cf88dd39a8ffe52eafde6f12bdc1bd8cf164a668a084dcb16a95ee925f) mandatory?
yes, would fail, i have no idea why... 

> 
>     * for a PR named `update dependencies, unsafe pip packages, fastapi` I would only expect changes to:
>       
>       * `poetry.lock`, `pyproject.toml` files
>       * some `imports` (maybe)


> > trying to fix this #1755 here aswell
> 
>     * bad idea. Keep the PR small.
> 
>     * are the refactorings in [lnbits/app.py](https://github.com/lnbits/lnbits/pull/1609/files#diff-e6e017cf88dd39a8ffe52eafde6f12bdc1bd8cf164a668a084dcb16a95ee925f) mandatory?
> 
>     * for a PR named `update dependencies, unsafe pip packages, fastapi` I would only expect changes to:
>       
>       * `poetry.lock`, `pyproject.toml` files
>       * some `imports` (maybe)

all changes are mandatory for this pr

**Review by jackstar12 — APPROVED** [review]

File: lnbits/app.py [hunk 2]
```diff
@@ -92,16 +96,16 @@ def create_app() -> FastAPI:
...
+    add_ip_block_middleware(app)
+    add_ratelimit_middleware(app)
+
     register_startup(app)
     register_routes(app)
     register_async_tasks(app)
     register_exception_handlers(app)
     register_shutdown(app)
 
-    # Allow registering new extensions routes without direct access to the `app` object
```
Comment (motorina0) [inline lnbits/app.py:101]: why refactor/move?

File: lnbits/core/views/generic.py (hunk 1 above)
Comment (jackstar12) [reply lnbits/core/views/generic.py:261]: getting the right error format with an `HTTPException` would require additional error formatters, I guess we'll leave it that way

**Review by arcbtc — APPROVED** [review]

**Pull request merged**